### ISSUE TITLE
[BUG] High RAM usage after updating to 0.2.6

### ISSUE THREAD

**Issue by kovdan01** [libre-tube/LibreTube#139]
**Describe the bug**

**To Reproduce**
1. Install LibreTube 0.2.6 from GitHub release section 
2. Watch videos for some time
3. See average RAM usage in settings is ~500MB (while ~120MB with LibreTube 0.2.4)
4. See maximum RAM usage in settings is ~1.5GB (while ~180MB with LibreTube 0.2.4)

**Smartphone (please complete the following information):**
 - Device: Asus Zenfone Max Pro M1 ZB602KL (RAM 3GB, storage 32GB)
 - Android version: 11 (LineageOS 18.1)

**Comment by lilcheti** [comment]
Oh seems like there is ram leakage then

**Comment by kovdan01** [comment]
@lilcheti Hello, are there any updates on the issue? The problem is still present in 0.2.7. Thanks!

**Comment by kovdan01** [comment]
Hello @lilcheti, tried v0.3.0 and the problem is still present. I decided to inspect changes [v0.2.4...v0.2.6 ](https://github.com/libre-tube/LibreTube/compare/v0.2.4...v0.2.6) and revealed that the first commit affected by the problem is ac28ef7e7fa7f2ac0b1ab04b1d1239af23ca343e (I have built my custom APK's for different commits and tested them). I'm not an Android developer so I can't reveal what's the problem with this commit.

Could you please look closer at the specified commit, reveal what's the root cause of the issue and fix it? High RAM usage makes several latest versions of your great app unusable on devices with 3-4GB of RAM. Thanks!

**Comment by kovdan01** [comment]
Hello @FireMasterK, maybe you have some thoughts regarding the issue? Looks like that it's present since ac28ef7e7fa7f2ac0b1ab04b1d1239af23ca343e (see details above). Thanks!

**Comment by Bnyro** [comment]
Cronet got implemented there which has improved the video loading times a lot. Undoing this would cause way longer loading times than it does now. Probably it requires more RAM to load the videos faster...

**Comment by kovdan01** [comment]
It would be cool if someone tries to detect if there's a memory leak because from my experience memory usage just keeps growing (from hundreds of MB to 1.5-2GB) and does not stop at any reasonable value. IMHO it's not very likely that cronet requires such amount of RAM. Unfortunately, I have almost no experience with Android development and absolutely no experience with Kotlin. If you could suggest some instruments that I can use to collect memory usage reports which might help you to detect possible leaks - please let me know.

**Comment by kovdan01** [comment]
I can also suggest a temporary workaround - give user a choice whether to use cronet or not (just add a checkbox in settings). I'm a bit afraid that it might hide a leak and nobody will return to the problem later (and a temporary workaround will become a persistent one), but if there are no complains - maybe the problem is not so critical or not a problem itself.

**Comment by Bnyro** [comment]
Disabling Cronet in the settings would probably be very hard, it would be better to find out what the reason for the ram leakage is.

**Comment by FireMasterK** [comment]
Can you see if https://nightly.link/libre-tube/LibreTube/actions/artifacts/253200329.zip fixes the issue?

**Comment by kovdan01** [comment]
Does not help unfortunately - RAM usage raises to >800MB in a couple of minutes :( Compared to v0.3.0 there are no RAM usage changes.

**Comment by Jomyy** [comment]
Maybe 

https://square.github.io/leakcanary/

Can help

**Comment by Bnyro** [comment]
@Jomyy I tried it but couldn't really find something noticeable.
I would appreciate if you could have a look into it if you find the time :)

**Comment by Jomyy** [comment]
Im not at home atm but i can try when im back

**Comment by Jomyy** [comment]
Im not at home atm but i can try when im back

**Comment by Bnyro** [comment]
I tried it with multiple devices, RAM usage was always less than 400mb. I also tried to find potential reasons with leakcanary but couldn't find anything. The issue might be device related, I'm not sure, but since you're apparently the only affected person I don't think we can do something about it.

**Comment by kovdan01** [comment]
@Bnyro Thanks for reply! I've monitored RAM usage for a couple of days and looks like that it's about 450MB at maximum - just as you said. I suppose that #139 helped and that was the root cause. Earlier I claimed that it didn't work - looks like that was related to debug build of the test APK that @FireMasterK have shared - because now on release builds RAM usage is normal.

Glad that the issue is fixed as for now and thanks for developing such a great app :)
